doquic.go

```diff
@@ -62,6 +62,9 @@ func (t *Transport) queryQUIC(ctx context.Context, addr *ServerAddr, query *dns.
 	if err != nil {
 		return nil, err
 	}
+	connPool.Add(closepool.CloserFunc(func() error {
+		return quicConn.CloseWithError(0x42, "quicConn closed!")
```

Comment:
Is `0x42` a ~standard error code for closing quic connections for DNS? Maybe you have already checked but, if not, I suggest we check whether the related RFC contains any pointers to known error codes. Also, in case we are using known error codes, I think we should probably comment about this or use named constants.

Reply:
I edited this to use `0x00`, which the DoQ RFC specifies as the "no error" way of closing a connection.